components/sidebar/sidebar_channel/index.js

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {getChannel, getChannelsNameMapInCurrentTeam} from 'mattermost-redux/selectors/entities/channels';
+import {getUserIdsInChannels, getUser} from 'mattermost-redux/selectors/entities/users';
+import {get as getPreference} from 'mattermost-redux/selectors/entities/preferences';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {leaveChannel} from 'mattermost-redux/actions/channels';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {Constants} from 'utils/constants.jsx';
+
+import SidebarChannel from './sidebar_channel.jsx';
+
+function mapStateToProps(state, ownProps) {
+    const config = getConfig(state);
+    const channel = getChannel(state, ownProps.channelId);
+    const tutorialStep = getPreference(state, Constants.Preferences.TUTORIAL_STEP, ownProps.currentUserId, 999);
+    const channelsByName = getChannelsNameMapInCurrentTeam(state);
+    const memberIds = getUserIdsInChannels(state);
+
+    let membersCount = 0;
+    if (memberIds && memberIds[channel.id]) {
+        membersCount = memberIds[channel.id].size;
+        if (memberIds[channel.id].has(ownProps.currentUserId)) {
+            membersCount--;
+        }
+    }
+
+    let unreadMentions = 0;
+    let unreadMsgs = 0;
+    const m = ownProps.membership;
+
+    if (channel.type === 'D') {
+        unreadMentions = channel.total_msg_count - m.msg_count;
+    } else if (m.mention_count > 0) {
+        unreadMentions = m.mention_count;
+    }
+    if (m.notify_props && m.notify_props.mark_unread !== 'mention' && channel.total_msg_count - m.msg_count > 0) {
+        unreadMsgs = 1;
+    }
+
+    let teammate = {}
```

Comment:
Would you also mind manually running ESLint on this file? There's a bug where it doesn't run on .js files, and this line is missing a semicolon, so it'll definitely trip it up

Reply:
Done